examples/PalmerPenguins.jl

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+using Pkg; Pkg.activate("../docs")
+using PalmerPenguins, Plots, DataFrames, Random
+using KernelFunctions, AbstractGPs, GPLikelihoods
+using EllipticalSliceSampling, Distributions
+df = dropmissing(DataFrame(PalmerPenguins.load()))
+df = df[randperm(nrow(df)), :]
+df[!,:flipper_length_mm] = convert(Vector{Float64}, df[!,:flipper_length_mm])
+df[!,:body_mass_g] = convert(Vector{Float64}, df[!,:body_mass_g])
+first(df, 5)
+
+# ## Binary classification using Bernoulli likelihood
+
+unique(df[!,:sex])
+
+
+x = RowVecs(Array(df[:, [
+                :bill_length_mm, 
+                :bill_depth_mm, 
+                :flipper_length_mm, 
+                :body_mass_g
+                ]]))
+y = df[!,:sex] .== "female"
+scatter(
+    df[!, :bill_length_mm],
+    df[!, :bill_depth_mm],
+    df[!, :flipper_length_mm],
+    group = df[!, :sex],
+    m = (0.5, :h, 5),
+)
+
+x_train, y_train = x[1:266], y[1:266]
+x_test, y_test = x[267:end], y[267:end];
+
+
+# logpdf without any parameters
+
+k = SqExponentialKernel()
+f = LatentGP(GP(k), BernoulliLikelihood(), 0.1)
+fx = f(x_test)
+logpdf(fx, (f=rand(fx.fx), y=y_test))
+
+
+function ℓ(params; x=x_train, y=y_train)
+    kernel = ScaledKernel(
+        KernelFunctions.transform(
+            SqExponentialKernel(),
+            ScaleTransform(exp(params[1]))
+        ),
+        exp(params[2])
+    )
+    f = LatentGP(GP(kernel), BernoulliLikelihood(), 0.1)
+    fx = f(x)
+    return logpdf(fx, (f=params[3:end], y=y))
+end
+
+prior = MvNormal(2+length(x_train), 1)
+ℓ(rand(prior)) # sanity check
+
+samples = sample(ESSModel(prior, ℓ), ESS(), 1_000; progress=true)
+samples_mat = reduce(hcat, samples);
+
+# +
+# source: https://github.com/tpapp/MCMCDiagnostics.jl/blob/master/src/MCMCDiagnostics.jl
+function autocorrelation(x::AbstractVector, k::Integer, v = var(x))
+    x1 = @view(x[1:(end-k)])
+    x2 = @view(x[(1+k):end])
+    V = sum((x1 .- x2).^2) / length(x1)
+    1 - V / (2*v)
+end
+
+function effective_sample_size(x::AbstractVector, v = var(x))
+    N = length(x)
+    τ_inv = 1 + 2 * autocorrelation(x, 1, v)
+    K = 2
+    while K < N - 2
+        Δ = autocorrelation(x, K, v) + autocorrelation(x, K + 1, v)
+        if Δ < 0
+            break
+        else
+            τ_inv += 2*Δ
+            K += 2
+        end
+    end
+    τ = min(1 / τ_inv, one(τ_inv))
+    τ * length(x)
+end
```

Comment:
IMO we should not reimplement existing functionality here but use MCMCDiagnostics or MCMCChains (implements different variants of ESS) directly.

Reply:
The idea was to avoid another dependency. But we might as well have this one as we are expected to use it frequently.